File: server/src/main/java/com/krillsson/sysapi/core/domain/storage/DiskLoad.java
package com.krillsson.sysapi.core.domain.storage;

public class DiskLoad {
    private final String name;
    private final DiskMetrics metrics;
    private final DiskSpeed speed;
    private final DiskHealth health;

    public DiskLoad(String name, DiskMetrics metrics, DiskSpeed speed, DiskHealth health) {
        this.name = name;
        this.metrics = metrics;
        this.speed = speed;
        this.health = health;
    }

    public String getName() {
        return name;
    }

    public DiskMetrics getMetrics() {
        return metrics;
    }

    public DiskSpeed getSpeed() {
        return speed;
    }

    public DiskHealth getHealth() {
        return health;
    }
}

File: server/src/main/java/com/krillsson/sysapi/core/domain/disk/DiskLoad.kt
package com.krillsson.sysapi.core.domain.disk

import com.krillsson.sysapi.core.domain.drives.DriveHealth
import com.krillsson.sysapi.core.domain.drives.DriveSpeed

data class DiskLoad(
    val name: String,
    val serial: String,
    val values: DiskValues,
    val speed: DriveSpeed,
    val health: DriveHealth
)